lib/wcc/contentful/model_builder.rb

```diff
@@ -35,67 +41,32 @@ def build_model(typedef)
             typedef
           end
 
-          define_singleton_method(:find) do |id, context = nil|
-            raw = WCC::Contentful::Model.store.find(id)
-            new(raw, context) if raw.present?
-          end
-
-          define_singleton_method(:find_all) do |filter = nil, context = nil|
-            if filter
-              filter.transform_keys! { |k| k.to_s.camelize(:lower) }
-              bad_fields = filter.keys.reject { |k| fields.include?(k) }
-              raise ArgumentError, "These fields do not exist: #{bad_fields}" unless bad_fields.empty?
-            end
-
-            query = WCC::Contentful::Model.store.find_all(content_type: content_type)
-            query = query.apply(filter) if filter
-            query.map { |r| new(r, context) }
-          end
-
-          define_singleton_method(:find_by) do |filter, context = nil|
-            filter.transform_keys! { |k| k.to_s.camelize(:lower) }
-            bad_fields = filter.keys.reject { |k| fields.include?(k) }
-            raise ArgumentError, "These fields do not exist: #{bad_fields}" unless bad_fields.empty?
-
-            result =
-              if defined?(context[:preview]) && context[:preview] == true
-                if WCC::Contentful::Model.preview_store.nil?
-                  raise ArgumentError,
-                    'You must include a contentful preview token in your WCC::Contentful.configure block'
-                end
-                WCC::Contentful::Model.preview_store.find_by(content_type: content_type, filter: filter)
-              else
-                WCC::Contentful::Model.store.find_by(content_type: content_type, filter: filter)
-              end
-
-            new(result, context) if result
-          end
-
-          define_singleton_method(:inherited) do |subclass|
-            # only register if it's not already registered
-            return if WCC::Contentful::Model.registered?(typedef.content_type)
-            WCC::Contentful::Model.register_for_content_type(typedef.content_type, klass: subclass)
-          end
-
           define_method(:initialize) do |raw, context = nil|
             ct = content_type_from_raw(raw)
             if ct != typedef.content_type
               raise ArgumentError, 'Wrong Content Type - ' \
                 "'#{raw.dig('sys', 'id')}' is a #{ct}, expected #{typedef.content_type}"
             end
-
-            @locale = context[:locale] if context.present?
-            @locale ||= 'en-US'
-            @id = raw.dig('sys', 'id')
-            @space = raw.dig('sys', 'space', 'sys', 'id')
-            @created_at = raw.dig('sys', 'createdAt')
-            @created_at = Time.parse(@created_at) if @created_at.present?
-            @updated_at = raw.dig('sys', 'updatedAt')
-            @updated_at = Time.parse(@updated_at) if @updated_at.present?
-            @revision = raw.dig('sys', 'revision')
+            @raw = raw.freeze
```

Comment:
do we want to expose raw as an `attr_reader`?

Reply:
Not sure - we don't have a use case for it yet, and since it's a deep hash `.freeze` doesn't protect us from someone going in and modifying it.  Then again, someone can always shoot themselves in the foot in ruby if they want to...